vpc/service/long_lived_lock.go

```diff
@@ -19,6 +24,81 @@ const (
 	timeBetweenTryingToAcquireLocks = 15 * time.Second
 )
 
+func (vpcService *vpcService) readLocks(rows *sql.Rows) ([]*vpcapi.Lock, error) {
+	locks := []*vpcapi.Lock{}
+	for rows.Next() {
+
+		lock := vpcapi.Lock{}
+		var heldUntil time.Time
+		err := rows.Scan(&lock.Id, &lock.LockName, &lock.HeldBy, &heldUntil)
+		if err != nil {
+			return nil, err
+		}
+
+		lock.HeldUntil, err = ptypes.TimestampProto(heldUntil)
+		if err != nil {
+			return nil, err
+		}
+
+		locks = append(locks, &lock)
+	}
+
+	return locks, nil
+}
+
+func (vpcService *vpcService) GetLocks(ctx context.Context, empty *empty.Empty) (*vpcapi.GetLocksResponse, error) {
+	rows, err := vpcService.db.QueryContext(ctx, "SELECT id, lock_name, held_by, held_until FROM long_lived_locks LIMIT 1000")
```

Comment:
Please wrap this in (read only) transaction

Reply:
Done but it's unclear to me what the intention of it is. As far as I understand it, the default transaction isolation is read committed so that prevents dirty reads. I'm probably missing something though...